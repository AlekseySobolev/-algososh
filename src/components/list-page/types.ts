export interface ILinkedList<T> {
  append: (element: T) => void;
  prepend: (element: T) => void;
  deleteHead: () => void;
  deleteTail: () => void;
  addByIndex: (element: T, index: number) => void;
  deleteByIndex: (index: number) => void;
  toArray: () => (T | string)[];
  changeByIndex: (index: number, newValue: string) => void;
}

export enum ListOperation {
  Prepend = "prepend",
  Append = "append",
  DeleteHead = "deleteHead",
  DeleteTail = "deleteTail",
  AddByIndex = "addByIndex",
  DeleteByIndex = "deleteByIndex"
}