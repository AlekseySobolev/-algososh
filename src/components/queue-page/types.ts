export interface IQueue<T> {
  enqueue: (item: T) => void;
  dequeue: () => void;
  peak: () => T | null;
  last: () => T | null;
  getHead: ()  => number;
  getTail: ()  => number;
  clean: () => void;
  getContainer: () => (T | null)[] | null;
}

export enum QueueOperation {
  Queue = "queue",
  Dequeue = "dequeue",
  Clean = "clean"
}