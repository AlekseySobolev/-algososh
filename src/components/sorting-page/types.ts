export interface Step {
    currentArray: number[];
    sortedIndexes: number[];
    aIndex?: number;
    bIndex?: number;
}

export enum typeOfSort {
    Select = "select",
    Bubble = "bubble"
}

export enum sortDirection {
    Ascending = "ascending",
    Descending = "descending"
}
