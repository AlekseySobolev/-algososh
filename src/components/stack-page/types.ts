export interface IStack<T> {
    push: (item: T) => void;
    pop: () => void;
    clear: () => void;
  }

export enum StackOperation{
    Push = "push",
    Pop = "pop",
    Clear = "clear"
}