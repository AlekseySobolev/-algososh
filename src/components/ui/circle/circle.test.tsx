import { fireEvent, render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import React from "react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";


describe("Circle tests",

    () => {

        it("Circle without letter", () => {

            const circle = <Circle />;
            render(circle);
            expect(screen.getByTestId('circle-box')).toMatchSnapshot();
  
        })

        it("Circle with letter", () => {
            let letter = 'h';
            const circle = <Circle letter= {letter}/>;
            render(circle);
            expect(screen.getByTestId('circle-box')).toHaveTextContent(letter);
        })

        it("Circle with head", () => {
            let head = 'head';
            const button = <Circle head={head} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toHaveTextContent(head);

        })

        it("Circle in head", () => {
            let head = 'head';
            const button = <Circle head={<Circle />} />;
            render(button);
            expect(screen.getAllByTestId('circle-box').length).toBe(2);

        })

        it("Circle with tail", () => {
            let tail = 'tail';
            const button = <Circle tail={tail} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toHaveTextContent(tail);

        })

        it("Circle in tail", () => {
            let tail = 'tail';
            const button = <Circle tail={<Circle />} />;
            render(button);
            expect(screen.getAllByTestId('circle-box').length).toBe(2);

        })

        it("Circle with index", () => {
            let index = 0;
            const button = <Circle index={index} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toHaveTextContent(String(index));

        })

        it("Circle with isSmall true", () => {
     
            const button = <Circle isSmall={true} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toMatchSnapshot();

        })

        it("Circle state equal Default", () => {
     
            const button = <Circle state={ElementStates.Default} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toMatchSnapshot();
          
        })
        it("Circle state equal Changing", () => {
     
            const button = <Circle state={ElementStates.Changing} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toMatchSnapshot();
        
        })
        it("Circle state equal Modified", () => {
     
            const button = <Circle state={ElementStates.Modified} />;
            render(button);
            expect(screen.getByTestId('circle-box')).toMatchSnapshot();
       
        })

    })
  
