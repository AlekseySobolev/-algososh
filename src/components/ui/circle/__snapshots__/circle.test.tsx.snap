// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle tests Circle state equal Changing 1`] = `
<div
  class="undefined "
  data-testid="circle-box"
>
  <div
    class="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    class="undefined   undefined"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    class="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle tests Circle state equal Default 1`] = `
<div
  class="undefined "
  data-testid="circle-box"
>
  <div
    class="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    class="undefined   undefined"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    class="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle tests Circle state equal Modified 1`] = `
<div
  class="undefined "
  data-testid="circle-box"
>
  <div
    class="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    class="undefined   undefined"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    class="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle tests Circle with isSmall true 1`] = `
<div
  class="undefined "
  data-testid="circle-box"
>
  <div
    class="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    class="undefined  undefined undefined"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    class="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;

exports[`Circle tests Circle without letter 1`] = `
<div
  class="undefined "
  data-testid="circle-box"
>
  <div
    class="text text_type_input text_color_input mb-4 undefined undefined undefined"
  />
  <div
    class="undefined   undefined"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input undefined"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 undefined undefined"
  />
  <div
    class="text text_type_input text_color_input mt-4 undefined undefined undefined"
  />
</div>
`;
